x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/TransportGetStackTracesAction.java

```diff
@@ -126,8 +130,31 @@ public TransportGetStackTracesAction(
     @Override
     protected void doExecute(Task submitTask, GetStackTracesRequest request, ActionListener<GetStackTracesResponse> submitListener) {
         licenseChecker.requireSupportedLicense();
-        StopWatch watch = new StopWatch("getResampledIndex");
         Client client = new ParentTaskAssigningClient(this.nodeClient, transportService.getLocalNode(), submitTask);
+
+        /*
```

Comment:
Ideally we would do this async on plugin startup (and after the data are available). This would avoid all locking and the first request also isn't penalized.